src/components/UserList.tsx

```diff
@@ -13,18 +20,43 @@ class User {
   }
 }
 
-interface UserListProp {
-  users: Array<User>;
-}
+const data = {
+  users: [
+    { id: '1', name: 'admin', role: 1 },
+    { id: '2', name: 'normal', role: 0 },
+  ],
+};
+
+const UserList: FC<RouteComponentProps> = ({ match }) => {
+  return (
+    <BrowserRouter>
+      <Switch>
+        <Route path={`${match.path}/:id`} component={UserListWithFilter} />
+        <Route path={`${match.path}`} component={AllUserList} />
+      </Switch>
+    </BrowserRouter>
+  );
+};
 
-const UserList: FC<UserListProp> = (props: UserListProp) => {
-  const users = props.users;
-  const listItems = users.map((user) => (
+const AllUserList: FC<{}> = () => {
+  const listItems = data.users.map((user) => (
     <li key={user.id}>
       {user.name}@{user.role}
     </li>
   ));
   return <ul>{listItems}</ul>;
 };
 
+const UserListWithFilter: FC<{}> = () => {
+  const { id } = useParams<{ id: string }>();
```

Comment:
a Hook method **useRouteMatch** is also available.
https://reactrouter.com/web/guides/quick-start/2nd-example-nested-routing